src/accelerate/utils/operations.py

```diff
@@ -351,6 +344,11 @@ def wrapper(*args, **kwargs):
             tensor = kwargs["tensor"]
         else:
             tensor = args[0]
+        if PartialState().device.type != tensor.device.type:
```

Comment:
`tensor` isn't guaranteed to be a tensor at this point, hence our failures. (e.g. could be a dict, list, etc). Let's use `find_device` instead so that it should work fine.

```suggestion
        if PartialState().device.type != find_device(tensor).type:
```